.github/workflows/ubuntu_x86_64.yml

```diff
@@ -0,0 +1,51 @@
+on: [pull_request]
+
+name: CI
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  RUST_BACKTRACE: 1
+
+jobs:
+  test-zig-rust-wasm:
+    name: test zig, rust, wasm
+    runs-on: [self-hosted, i5-4690K]
+    timeout-minutes: 90
+    env:
+      RUSTC_WRAPPER: /home/small-ci-user/.cargo/bin/sccache
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          clean: "true"
+
+      - name: zig fmt check, zig tests
+        run: cd crates/compiler/builtins/bitcode && ./run-tests.sh
+
+      - name: zig wasm tests
+        run: cd crates/compiler/builtins/bitcode && ./run-wasm-tests.sh
+
+      - name: regular rust tests
+        run: cargo test --locked --release --features with_sound serde --workspace && sccache --show-stats
+
+      - name: test the dev and wasm backend # these tests require an explicit feature flag
+        run: cargo test --locked --release --package test_gen --no-default-features --features gen-dev && sccache --show-stats
+
+      - name: test gen-wasm single threaded # gen-wasm has some multithreading problems to do with the wasmer runtime
+        run: cargo test --locked --release --package test_gen --no-default-features --features gen-wasm -- --test-threads=1 && sccache --show-stats
+
+      - name: run `roc test` on Str builtins
+        run: cargo run --locked --release -- test crates/compiler/builtins/roc/Str.roc && sccache --show-stats
+
+      #TODO pass --locked into the script here as well, this avoids rebuilding dependencies unnecessarily
+      - name: wasm repl test
+        run: crates/repl_test/test_wasm.sh && sccache --show-stats
+
+      #TODO i386 (32-bit linux) cli tests
+
+      #TODO verify-no-git-changes
```

Comment:
Still have some TODO's but I want to get this merged in because many of the above tests are currently not being run on PRs.